imageroot/actions/configure-module/81reports_api

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+
+#
+# Copyright (C) 2023 Nethesis S.r.l.
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+
+"""
+This file is created with separation of resposability in mind.
+This configures API segment of reports.
+"""
+
+import json
+import sys
+import os
+
+import agent
+from jinja2 import Environment, FileSystemLoader
+
+# Create folder scaffolding
+os.makedirs('report', exist_ok=True)
+
+# Fetch request
+request = json.load(sys.stdin)
+
+# Set environment variables
+agent.set_env(
+    'REPORTS_INTERNATIONAL_PREFIX',
+    request['reports_international_prefix']
+)
+
+# Create configuration file
+env = Environment(
+    loader=FileSystemLoader('../templates/'),
+    autoescape=True
+)
+
+template = env.get_template('reports-api.j2')
+with open('./report/api-config.json', 'w', encoding='utf-8') as file:
+    file.write(template.render({
+        'reports_api_port': os.environ['REPORTS_API_PORT'],
+        'reports_redis_port': os.environ['REPORTS_REDIS_PORT'],
+        'nethvoice_mariadb_port': os.environ['NETHVOICE_MARIADB_PORT'],
+        'reports_password': os.environ['REPORTS_PASSWORD'],
+        'reports_secret': os.environ['REPORTS_SECRET'],
+        'reports_api_endpoint': 'https://' + os.environ['NETHVOICE_HOST']
+        + '/pbx-report-api',
+        'reports_api_key': os.environ['REPORTS_API_KEY'],
+    }))
```

Comment:
I'm not sure I like this approach: why not generating the config inside the container like other apps?
Otherwise, if you want to use a template, you can expand it in a `ExecPre` command inside the systemd unit.

Reply:
Because putting heavy configuration printing in entrypoint makes it unreadable (and error prone), putting the config generation inside the systemd doesn't feel right because it would've been generated every run (if it doesn't change, why bother?).

Putting config file generation during the configuration of the module makes the generation of the file only when something might have changed in the config, also, separating the config file from the entrypoint (should) avoid rebuilding of the image due to no changes to the codebase

Reply:
@Stell0 what do you think?